src/sync.h

```diff
@@ -138,14 +135,14 @@ class SCOPED_LOCKABLE UniqueLock : public Base
     void Enter(const char* pszName, const char* pszFile, int nLine)
     {
         EnterCritical(pszName, pszFile, nLine, Base::mutex());
-#ifdef DEBUG_LOCKCONTENTION
-        if (!Base::try_lock()) {
-            PrintLockContention(pszName, pszFile, nLine);
-#endif
+        if (LogAcceptCategory(BCLog::LOCK)) {
+            if (!Base::try_lock()) {
+                LogPrintf("Lock contention: %s, %s:%d\n", pszName, pszFile, nLine);
+                Base::lock();
+            }
+        } else {
```

Comment:
```suggestion
        if (LogAcceptCategory(BCLog::LOCK) && !Base::try_lock()) {
            LogPrintf("Lock contention: %s, %s:%d\n", pszName, pszFile, nLine);
        }
```

?

Reply:
Thanks. If I understand the suggestion, this would only lock if the lock logging is on and `try_lock()` succeeds.

Reply:
```cpp
    void Enter(const char* pszName, const char* pszFile, int nLine)
    {
        EnterCritical(pszName, pszFile, nLine, Base::mutex());
        if (LogAcceptCategory(BCLog::LOCK) && !Base::try_lock()) {
            LogPrintf("Lock contention: %s, %s:%d\n", pszName, pszFile, nLine);
        }
        Base::lock();
    }
```

Reply:
The same behavior as with your current code.

Reply:
nm, my suggestion is wrong.

Reply:
Thank u